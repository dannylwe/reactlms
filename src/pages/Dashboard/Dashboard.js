import React, { Component } from "react";
import axios from "axios";
import "./Dashboard.css";
import NavBar from "../../components/NavBar/NavBar";

class Dashboard extends Component {
  state = {
    items: ""
  };

  componentDidMount() {
    this.getParcels();
  }

  getParcels() {
    return axios
      .get("https://challenge3andela.herokuapp.com/api/v1/parcels", {
        withCredentials: true
      })
      .then(res => {
        this.mtd(res.data);
      });
  }
  mtd(data) {
    this.setState({ items: data["item info"] });
  }

  render() {
    const { items } = this.state;

    const itemValues =
      items.length > 0
        ? items.map(item => (
            <tr>
              <th>{item.destination}</th>
              <th>{item.nickname}</th>
              <th>{item.status}</th>
              <th>{item.pickup}</th>
            </tr>
          ))
        : "loading......";
    return (
      <>
        <NavBar />
        <br />
        <table>
          <tbody>
            <tr>
              <th>Destination</th>
              <th>Order Nickname</th>
              <th>Status</th>
              <th>Pickup Location</th>
            </tr>
            {itemValues}
          </tbody>
        </table>
      </>
    );
  }
}

export default Dashboard;
