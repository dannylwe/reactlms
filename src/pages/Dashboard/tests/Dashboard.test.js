import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import { mount, shallow } from "enzyme";
import Dashboard from "../Dashboard";

describe("Dashboard", () => {
  it("should render without failing", () => {
    const wrapper = renderer.create(
      <MemoryRouter>
        <Dashboard />
      </MemoryRouter>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
describe("Dashboard", () => {
  it("calls componentDidMount", () => {
    const spy = jest.spyOn(Dashboard.prototype, "componentDidMount");
    const wrapper = mount(
      <MemoryRouter>
        <Dashboard />
      </MemoryRouter>
    );
    wrapper.instance().componentDidMount();
    expect(spy).toHaveBeenCalled();
  });
  it("calls getParcels", () => {
    const component = shallow(<Dashboard />);
    const getParcels = jest.fn().mockImplementation();
    getParcels.mockReturnValue([{ key: 0, itemName: "parcel" }]);
    component.instance().getParcels()
  });
});
