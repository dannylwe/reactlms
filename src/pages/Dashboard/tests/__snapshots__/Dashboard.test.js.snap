// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should render without failing 1`] = `
Array [
  <nav
    className="navbar navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/dashboard"
    >
      Home
    </a>
    <a
      className="navbar-brand"
      href="/login"
      onClick={[Function]}
    >
      Logout
    </a>
  </nav>,
  <br />,
  <table>
    <tbody>
      <tr>
        <th>
          Destination
        </th>
        <th>
          Order Nickname
        </th>
        <th>
          Status
        </th>
        <th>
          Pickup Location
        </th>
      </tr>
      You do not have any posts
    </tbody>
  </table>,
]
`;
